lib/utils/thingMetadata.js

```diff
@@ -0,0 +1,16 @@
+import { ajax } from '../environment';
+import { batch } from './async';
+
+export const getPostMetadata = batch(async requests => {
+	const byId = requests.map(r => r.id)
+		.map(id => `t3_${id}`).join(',');
+
+	const { data: { children } } = await ajax({
+		url: `/by_id/${byId}.json`,
+		type: 'json',
+	});
+
+	return children.reduce((map, { data }) => map.set(data.id, data), new Map());
```

Comment:
I think you should be able to `return children.map(c => c.data)`, since results are returned in order (and not deduped, as you've seen 😜 )